### ISSUE TITLE
spring boot 2.2.2.RELEASE not work

### ISSUE THREAD

**Issue by beykery** [akkinoc/logback-access-spring-boot-starter#55]
when i use spring boot 2.2.2.RELEASE  version , it thow exception like this :

The following method did not exist:

    org.springframework.boot.web.embedded.undertow.ConfigurableUndertowWebServerFactory.addDeploymentInfoCustomizers([Lorg/springframework/boot/web/embedded/undertow/UndertowDeploymentInfoCustomizer;)V

the method addDeploymentInfoCustomizers of ConfigurableUndertowWebServerFactory don't exist !

**Comment by istonikula** [comment]
It seems ConfigurableUndertowWebServerFactory#addDeploymentInfoCustomizers has been deprecated since spring-boot@2.1.7  

More info
- https://github.com/spring-projects/spring-boot/issues/17562
- https://github.com/spring-projects/spring-boot/issues/17555

Also. there seems to be a new preferred way to apply customizers (since spring-boot@2.0.0.M3)
- https://github.com/spring-projects/spring-boot/issues/16514
- https://github.com/spring-projects/spring-boot/pull/16584

**Comment by beykery** [comment]
so, do you have any plan to upgrade this project with spring boot 2.x dependence ?
this project have not updated from 2 years ago .

**Comment by alexanderjohn** [comment]
@beykery :

I was able to build a work around using the new way of applying Customizers applied in fix: https://github.com/spring-projects/spring-boot/pull/16584

Just drop this class in to your config and it will turn off the `UndertowLogbackAccessInstaller` and correctly load the customizers into the `UndertowServletWebServerFactory`

```
@Slf4j
@RequiredArgsConstructor
@Configuration
public class LogbackAccessConfig {

    private final LogbackAccessProperties logbackAccessProperties;
    private final Environment environment;
    private final ApplicationEventPublisher applicationEventPublisher;

    @Bean
    public UndertowLogbackAccessInstaller undertowLogbackAccessInstaller(
            ConfigurableUndertowWebServerFactory container) {
        logger.debug("Installed Logback-access: container=[{}]", container);
        return null;
    }

    @Bean
    public UndertowBuilderCustomizer logbackAccessBuilderCustomizer() {
        return builderCustomizer -> {
            builderCustomizer.setServerOption(UndertowOptions.RECORD_REQUEST_START_TIME, true);
        };
    }

    @Bean
    public UndertowDeploymentInfoCustomizer logbackAccessDeploymentInfoCustomizer() {
        return deploymentInfo -> {
            deploymentInfo.addInitialHandlerChainWrapper(handler -> new LogbackAccessUndertowHttpHandler(
                    logbackAccessProperties, environment, applicationEventPublisher, handler));
        };
    }
}```

**Comment by beykery** [comment]
thank you .

**Comment by akkinoc** [comment]
Sorry for the late reply. Thanks for report!
Fixed in PR #57.
